test/fast/Unit tests/nvm_download

```diff
@@ -15,4 +15,9 @@ nvm_download "https://raw.githubusercontent.com/nvm-sh/nvm/HEAD/install.sh" >/de
 # nvm_download should fail to download wrong_install.sh
 ! nvm_download "https://raw.githubusercontent.com/nvm-sh/nvm/HEAD/wrong_install.sh" >/dev/null || die "nvm_download should fail to download no existing file"
 
+# nvm_download should pass when calling with auth header
+NVM_AUTH_HEADER="Authorization: Bearer test-token"
+nvm_download "https://httpbin.org/bearer" >/dev/null || die "nvm_download unable to send auth header"
```

Comment:
love it.

if it turns out to be unreliable, seems like we can `docker pull kennethreitz/httpbin && docker run -p 80:80 kennethreitz/httpbin` and hit it locally.